### ISSUE TITLE
Q-RDFUnit: UnifiedViews is missing Jena support 

### ISSUE THREAD

**Issue by anisaR** [UnifiedViews/Plugins-QualityAssessment#9]
RDFUnit tool is based on Jena library. When the dpu is compiled this dpu will require all the dependencies of Jena in UV. Although some dependencies are satisfied, meaning that they are included in the pom file (e.g. com.hp.hpl.jena.rdf.model), it recursively requires other dependencies. This implies the increasing size of the dpu and it doesn't seem that the inclusion of the dependencies will end because it requires other dependencies rather than the Jena's ones. 

Since in local environment the tool works, we think that these libraries should be included in UV. 

Any suggestion about this issue?

**Comment by tomas-knap** [comment]
Please discuss this with @skodape. In general the problem you discribe is how to bundle Jena library.

**Comment by AndreAga** [comment]
I wrote to @skodape . He will discuss with you @tomas-knap . I wait an answer when Petr will finish his tasks.

**Comment by tomas-knap** [comment]
@AndreAga So the problem is that you are not able to load the prepared DPU to frontend, because certain (transitive) dependency is missing, am I right?

**Comment by AndreAga** [comment]
@tomas-knap Yep, I added Jena OSGi support and now the problem is with another dependance. You can check the pom and try.

**Comment by tomas-knap** [comment]
Btw. I was checking the RDFUnit - using the demo http://rdfunit.aksw.org/demo/ and it seems to me that the automatically generated test cases are useless. Because they are generating basically issues in the vocabularies not issues related to the usage of the vocabulary in the data (see below example). 

So it seems to me that the only reason for which we can use it are custom test cases - but custom test cases are basically sparql queries - sets of sparql queries. And for this, you do not need RDFUnit, but we can prepare DPU, which is able to execute set of SPARQL ASK queries. 

At least this is my opinion, what do you think? @anisaR?

For data
```
<http://linked.opendata.cz/resource/court/cz/krajsky-soud-v-brne>
	 a court-kinds:Regional, gr:BusinessEntity, s:Organization ;
	 dcterms:title "Krajský soud v Brně"@cs ;
	 adms:identifier <http://linked.opendata.cz/resource/court/cz/krajsky-soud-v-brne/identifier/1> ;
	 s:address <http://linked.opendata.cz/resource/court/cz/krajsky-soud-v-brne/postal-address/1> ;
	 s:telephone "+420546511111"^^xsd:string ;
	 s:faxNumber "+420546513362"^^xsd:string ;
	 owl:sameAs <http://linked.opendata.cz/resource/business-entity/CZ00215724> ;
	 s:email "podatelna@ksoud.brn.justice.cz"^^xsd:string ;
	 s:url <http://portal.justice.cz/justice2/soud/soud.aspx?j=26&o=16&k=310> 
	 .
```
I got hundreds/thousands of the following errors: 

<http://rdfunit.aksw.org/data/results#43617da1-2ace-11b2-8045-d4ae52e56c05/aab41814-2ace-11b2-8045-d4ae52e56c05>
        a                     rut:AggregatedTestResult , rut:TestCaseResult , rut:StatusTestCaseResult ;
        dcterms:date          "2015-04-01T15:06:38.87Z"^^xsd:dateTime ;
        dcterms:description   "http://purl.org/goodrelations/v1#hasMaxValueInteger has rdfs:domain different from: http://purl.org/goodrelations/v1#QuantitativeValueInteger" ;
        rut:resultCount       0 ;
        rut:resultPrevalence  0 ;
        rut:resultStatus      rut:ResultStatusSuccess ;
        rut:testCase          rutt:gr-RDFSDOMAIN-184621b0bfea59b6892b9b5c6b6f7a49a4cab64b2512cb568ea2df1a40e062f2 ;
        rut:testCaseLogLevel  rlog:ERROR ;
        prov:wasGeneratedBy   rutr:43617da1-2ace-11b2-8045-d4ae52e56c05 .

<http://rdfunit.aksw.org/data/results#43617da1-2ace-11b2-8045-d4ae52e56c05/aab503e4-2ace-11b2-8045-d4ae52e56c05>
        a                     rut:StatusTestCaseResult , rut:AggregatedTestResult , rut:TestCaseResult ;
        dcterms:date          "2015-04-01T15:06:50.822Z"^^xsd:dateTime ;
        dcterms:description   "Class http://purl.org/goodrelations/v1#Location is owl:disjointWith with http://purl.org/goodrelations/v1#WarrantyPromise" ;
        rut:resultCount       0 ;
        rut:resultPrevalence  -1 ;
        rut:resultStatus      rut:ResultStatusSuccess ;
        rut:testCase          rutt:gr-OWLDISJC-3e928fa7cb95707b5d77d40e3a573b9718f44ca372a9aa8ac79560fa4b7754cc ;
        rut:testCaseLogLevel  rlog:ERROR ;
        prov:wasGeneratedBy   rutr:43617da1-2ace-11b2-8045-d4ae52e56c05 .

<http://rdfunit.aksw.org/data/results#43617da1-2ace-11b2-8045-d4ae52e56c05/aab6b11c-2ace-11b2-8045-d4ae52e56c05>
        a                     rut:StatusTestCaseResult , rut:TestCaseResult , rut:AggregatedTestResult ;
        dcterms:date          "2015-04-01T15:06:55.434Z"^^xsd:dateTime ;
        dcterms:description   "http://schema.org/editor has different range from: http://schema.org/Person" ;
        rut:resultCount       0 ;
        rut:resultPrevalence  0 ;
        rut:resultStatus      rut:ResultStatusSuccess ;
        rut:testCase          rutt:schema-RDFSRANGE-c17f3c1f851cd8cb7e01bdf316169c51883b104df4ba01d833da9a63fca69de8 ;
        rut:testCaseLogLevel  rlog:ERROR ;
        prov:wasGeneratedBy   rutr:43617da1-2ace-11b2-8045-d4ae52e56c05 .

<http://rdfunit.aksw.org/data/results#43617da1-2ace-11b2-8045-d4ae52e56c05/aab7cd68-2ace-11b2-8045-d4ae52e56c05>
        a                     rut:TestCaseResult , rut:AggregatedTestResult , rut:StatusTestCaseResult ;
        dcterms:date          "2015-04-01T15:07:02.832Z"^^xsd:dateTime ;
        dcterms:description   "http://schema.org/eventStatus does not have defined range: http://schema.org/EventStatusType" ;
        rut:resultCount       0 ;
        rut:resultPrevalence  0 ;
        rut:resultStatus      rut:ResultStatusSuccess ;
        rut:testCase          rutt:schema-RDFSRANGE-MISS-c405679a3b4e2dbcbee71dba251d63f8c7838d78df1764d441e1794ffa486b65 ;
        rut:testCaseLogLevel  rlog:WARN ;
        prov:wasGeneratedBy   rutr:43617da1-2ace-11b2-8045-d4ae52e56c05 .

**Comment by anisaR** [comment]
No, it is not like it seems. I would suggest to load a dataset and there is an important point to consider while loading a dataset. Most of the vocabularies used in the dataset should be defined somewhere. In particular, RDFUnit is based on this service http://lov.okfn.org/dataset/lov/ which has defined almost all the vocabularies. What we are missing is our own vocabulary used inside our project which is not there. In this case, after doing an analysis, in the configuration of the dpu we propose to add an option for the user to upload its own vocabulary or to insert the link of its own ontology. 
In this way, you will be able to generate errors that are directly related to the data. The point is that in the demo version it generates everything but in the local version we select only those errors that are generated by the dataset itself. 

I see a lot of benefits in using this tool rather than using other validators. I would say that we have done a deeper analysis and we think that this tool is more powerful than RDFAlert. However, in order to finish all the implementation and real usage in UV we need your  @tomas-knap  or @skodape support. 

First, try with a dataset from comsode and second help Andrea and Vincenzo to load this dpu in UV.

**Comment by AndreAga** [comment]
@tomas-knap @skodape @anisaR I've just commit the installation steps.

**Comment by tomas-knap** [comment]
Anisa, can you point me to some prepared pipelines, which I can just use to see the output of RDFUnit? Because I am still not persuaded that the report will be somehow useful if it produces tons of problems, mainly related to inconsistencies in the vocabularies (which I am not interested in)

**Comment by tomas-knap** [comment]
@AndreAga The instalation is too complicated for normal user - the required JAR file should be prepared and published somewhere,e.g., on the maven server which github may host (http://kwebble.com/blog/2014/02/19/use-github-to-host-your-own-maven-repo). If we publish required JAR file directly, the installation steps could be heavily simplified, right?

**Comment by tomas-knap** [comment]
@anisaR Please provide me with a sample COMSODE pipeline where the RDFUnit is proved to be usefull. Otherwise I do not see any point in using RDFUnit. I see it as a good tool for custom policies but rather poor, not usefull tool for automatically generated policies. But using it just for custom policies, much simpler approach may be used (textarea where you can define couple of SPARQL ask queries) to start with, which will solve the same goal.

**Comment by AndreAga** [comment]
@tomas-knap some changes have been made in the last days in the repo of rdfunit tool. 
I think you can now simply clone this repo: https://github.com/AKSW/RDFUnit.git and run mvn install to get rdfunit tool working.

The other steps, about RDFUnit DPU, are temporary in order to get an example of output :)

**Comment by AndreAga** [comment]
Hi @tomas-knap, @anisaR said to me that you want a simple DPU using the jar of RDFUnit tool. The fact is that this .jar doesn't exist. The command line version of this tool uses a unix executable file (bin/rdfunit), so if you want a new DPU, the installation of this tool must be present on the server. Honestly I don't know if this is the best solution, but anyway I wanted to let you know.

**Comment by tomas-knap** [comment]
Well, depending on the usability issues with existing Q-ACC1 we will decide what next
